engine/channels.go

```diff
@@ -63,10 +65,9 @@ func UniqueChannels(channels network.ChannelList) network.ChannelList {
 	// has already been added to uniques.
 	// We use identifier of RoleList to determine its uniqueness.
 	for _, channel := range channels {
-		id := channelRoleMap[channel].ID()
-
 		// non-cluster channel deduplicated based identifier of role list
 		if _, cluster := ClusterChannel(channel); !cluster {
+			id := channelRoleMap[channel].ID()
```

Comment:
The old place is already inside the loop 🧐

I have only placed it inside the `if`, which is only saving computation because it will be computed less often.